records_mover/records/job/config.py

```diff
@@ -2,7 +2,7 @@
 import inspect
 from typing import Any, Dict, List, Callable
```

Comment:
The below logic is all kinds of weird.  I get that.

Here's what's going on:

Records Mover is both a Python library and a command line tool.

In terms of the library, the richest surface area of the API are source and target factory methods - where you specify what source and target you want, and configure them exactly as you want:
* https://github.com/bluelabsio/records-mover/blob/master/records_mover/records/sources/factory.py
* https://github.com/bluelabsio/records-mover/blob/master/records_mover/records/targets/factory.py

The naive way to write a command line parser would be to take each of these functions and write an [argparse](https://docs.python.org/3/library/argparse.html) parser.  You'd certainly want to do something at least mildly clever to combine those, so you don't have to rewrite the "file2" part of "file2url" and "file2table", etc, or you're dealing with exponential growth as sources and targets are added.

On top of that, it turns out that a lot of those parameters to methods are common - things like the records_format, the db_engine, the initial_hints, etc.  Copying and pasting those over and over is kind of silly.

It's also true that a lot of those are pretty simple - strings, booleans, etc, possibly with default values.  And they already have documentation written.

So I do something more clever, and I use Python's reflection abilities to pull parameter lists, types and documentation and generate `argparse` parsers.  There's even an intermediate form as JSON schema (we used to have a layer which would accept records mover instructions as a JSON-style request).   You can draw your own conclusion as to whether 'clever' is a positive description or not.

Aaaanyway....

![image](https://user-images.githubusercontent.com/14927757/98750044-f6529b80-238a-11eb-99c0-fc50e8e043f2.png)